### PR TITLE
refactor(mastodon): remove redundant css & use vars

### PR DESCRIPTION
**Pull request by isabelroses**
## 🔧 What does this fix? 🔧

Cleanup time for mastodon, this has been a long time coming.

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [contributing guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/CONTRIBUTING.md).
- [x] I have updated the version appropriately in the `==UserStyle==` header of the `catppuccin.user.css` file.

### CODE REVIEW

File: styles/mastodon/catppuccin.user.css [hunk 1]
```diff
@@ -537,7 +531,7 @@ domain("wetdry.world") {
       border-color: @crust;
     }
 
-    .poll__option input[type="text"]:focus,
+    .poll__option input[type=text]:focus,
```
Comment (uncenter) [inline styles/mastodon/catppuccin.user.css:534]:
```suggestion
    .poll__option input[type="text"]:focus,
```

File: styles/mastodon/catppuccin.user.css [hunk 2]
```diff
@@ -423,18 +416,19 @@ domain("wetdry.world") {
     .button.logo-button:active,
     .button.logo-button:focus,
     .button.logo-button:hover,
+    .notification__filter-bar button.active:before,
```
Comment (uncenter) [inline styles/mastodon/catppuccin.user.css:419]:
```suggestion
    .notification__filter-bar button.active::before,
```

File: styles/mastodon/catppuccin.user.css [hunk 3]
```diff
@@ -183,10 +186,77 @@ domain("wetdry.world") {
...
+    }
+
+    .privacy-dropdown__option.active {
+      background: @accent-color !important;
+      .privacy-dropdown__option__content,
+      strong,
+      i {
+        color: @crust;
+      }
+    }
```
Comment (uncenter) [inline styles/mastodon/catppuccin.user.css:219]:
```suggestion
    .privacy-dropdown__option {
      background: @surface0;

      .privacy-dropdown__option__content,
      strong,
      i {
        color: @text;
      }

      &:hover, &.active {
        background: @accent-color !important;

        .privacy-dropdown__option__content,
        strong,
        i {
          color: @crust;
        }
      }
    }
```

File: styles/mastodon/catppuccin.user.css [hunk 4]
```diff
@@ -183,10 +186,77 @@ domain("wetdry.world") {
...
+
+    .language-dropdown__dropdown__results__item.active {
+      background: @accent-color;
+      > span {
+        color: @crust;
+      }
+    }
```
Comment (uncenter) [inline styles/mastodon/catppuccin.user.css:258]:
```suggestion
    .language-dropdown__dropdown__results__item {
      > span {
        color: @text;
      }

      &:hover, &.active {
        background: @accent-color;

        > span {
          color: @crust;
        }
      }
```

File: styles/mastodon/catppuccin.user.css [hunk 5]
```diff
@@ -423,25 +473,57 @@ domain("wetdry.world") {
...
+
+    .autosuggest-textarea__suggestions__item {
+      .account.small .display-name {
+        span {
+          color: @text !important;
+        }
+      }
+    }
```
Comment (uncenter) [inline styles/mastodon/catppuccin.user.css:522]:
```suggestion
    .autosuggest-textarea__suggestions__item {
      .account.small .display-name {
        span {
          color: @text !important;
        }
      }

      &.selected, &:hover {
        background: @accent-color !important;

        .account.small .display-name {
          span,  bdi > strong {
            color: @crust !important;
          }
        }
      }
```

**Comment by uncenter** [comment]
I think we should merge this as is and feel free to open further PRs?

**Comment by isabelroses** [comment]
> I think we should merge this as is and feel free to open further PRs?

Yup sounds good to me.

**Review by uncenter — APPROVED** [review]

**Pull request merged**